File: pkgs/spatialdb/tests/libtests/TestSpatialDB.cc
// -*- C++ -*-
//
// ----------------------------------------------------------------------
//
//                           Brad T. Aagaard
//                        U.S. Geological Survey
//
// {LicenseText}
//
// ----------------------------------------------------------------------
//

#include <portinfo>

#include "TestSpatialDB.h" // implementation of class methods

#include "spatialdata/spatialdb/SpatialDB.h" // USES SimpleDB
#include "spatialdata/spatialdb/SimpleDB.h" // USES SimpleDB

#include "spatialdata/spatialdb/SimpleIO.h" // USES SimpleIOAscii
#include "spatialdata/spatialdb/SimpleIOAscii.h" // USES SimpleIOAscii

#include "spatialdata/geocoords/CoordSys.h" // USES CSCart
#include "spatialdata/geocoords/CSCart.h" // USES CSCart

#if defined(HAVE_PYTHIA)
#include "journal/firewall.h" // USES FIREWALL
#include "pythiautil/FireWallUtil.h" // USES FIREWALL
#else
#include <assert.h>
#define FIREWALL assert
#endif

// ----------------------------------------------------------------------
CPPUNIT_TEST_SUITE_REGISTRATION( spatialdata::spatialdb::TestSpatialDB );

// ----------------------------------------------------------------------
// Setup test subject
void
spatialdata::spatialdb::TestSpatialDB::setUp(void)
{ // setUp
  SimpleDB* pDB = new SimpleDB;

  const char* filename = "data/spatialdb.dat";
  SimpleIOAscii iohandler;
  iohandler.filename(filename);
  pDB->ioHandler(&iohandler);

  _pDB = pDB;
  _pDB->open();

  pDB->queryType(SimpleDB::NEAREST);
} // setUp

// ----------------------------------------------------------------------
// Cleanup test subject
void
spatialdata::spatialdb::TestSpatialDB::tearDown(void)
{ // tearDown
  delete _pDB; _pDB = 0;
} // tearDown

// ----------------------------------------------------------------------
// Test SpatialDB
void
spatialdata::spatialdb::TestSpatialDB::testDB(void)
{ // testDB
  FIREWALL(0 != _pDB);

  const char* names[] = {"two", "one"};
  const char* units[] = {"m", "m"};
  const int numVals = 2;
  const double queryLoc[] = { 1.0, 2.0, 3.0 };
  const double pVals[] = { 6.3, 4.7 };

  _pDB->queryVals(names, numVals);

  double* pValsQ = (0 < numVals) ? new double[numVals] : 0;
  spatialdata::geocoords::CSCart csCart;
  csCart.initialize();
  _pDB->query(&pValsQ, numVals, queryLoc[0], queryLoc[1], queryLoc[2],
	      &csCart);
  const double tolerance = 1.0e-06;
  for (int iVal=0; iVal < numVals; ++iVal)
    CPPUNIT_ASSERT_DOUBLES_EQUAL(pValsQ[iVal]/pVals[numVals-iVal-1],
				 1.0, tolerance);

  delete[] pValsQ; pValsQ = 0;
  _pDB->close();
} // testDB

// version
// $Id: TestSpatialDB.cc,v 1.1 2005/05/25 18:43:06 baagaard Exp $

// End of file 

File: pkgs/spatialdb/tests/langtests/TestSpatialDB.py
#!/usr/bin/env python
#
# ======================================================================
#
#                           Brad T. Aagaard
#                        U.S. Geological Survey
#
# {LicenseText}
#
# ======================================================================
#

import unittest

class TestSpatialDB(unittest.TestCase):

  def setUp(self):

    from spatialdata.spatialdb.SimpleIOAscii import SimpleIOAscii
    iohandler = SimpleIOAscii()
    iohandler.inventory.filename = "data/spatialdb.dat"
    iohandler.initialize()

    from spatialdata.spatialdb.SimpleDB import SimpleDB
    db = SimpleDB()
    db.query_type = "Nearest"
    db.iohandler = iohandler
    db.initialize()

    self._db = db

    from spatialdata.geocoords.CSCart import CSCart
    cs = CSCart()
    cs.initialize()
    self._cs = cs
    return

  def test_cquery(self):
    self._db.open()
    self._db.queryVals(["two", "one"])
    self._cquery()
    self._db.close()
    
    return

  def test_f77query(self):
    self._db.open()
    self._db.queryVals(["two", "one"])
    self._f77query()
    self._db.close()
    
    return

  def _cquery(self):
    import langtestspatialdb as bindings
    bindings.cTestQuery(self._db.handle(), self._cs.handle())
    return

  def _f77query(self):
    import langtestspatialdb as bindings
    bindings.f77TestQuery(self._db.handle(), self._cs.handle())
    return

# version
__id__ = "$Id$"

# End of file 
